Update READMEs and links for Theme Kit

mcvinci opened the pull request:
## This PR:
- Updates the main README to align with information on shopify.dev.
- Updates the "Contributing to Theme Kit" README to improve clarity, clean up some language, and provide additional information about Go and Node development.
- Updates the "Releasing" README to improve clarity and update procedures around documentation.
- Updates several links to point to information on shopify.dev.
- Moves the python script out of the `/docs` folder and removes the `/docs` folder
- Relates to https://github.com/Shopify/shopify-dev/pull/5070 and https://github.com/Shopify/shopify-dev/issues/5032.

cc: @gfscott 

### Tophatted

#### Main README

![screencapture-github-Shopify-themekit-blob-93d8d6f0698fd1dbd8f262592ad3a99c203d5681-README-md-2021-02-02-14_14_17](https://user-images.githubusercontent.com/65373971/106670048-65dac000-6561-11eb-9163-6279b1c3d91b.png)

#### Manual installation README

![screencapture-github-Shopify-themekit-blob-35cef3089f60a8d2bc02ddc2c30a0e5c4a416a97-MANUAL-INSTALL-md-2021-02-02-14_16_59](https://user-images.githubusercontent.com/65373971/106670110-768b3600-6561-11eb-924c-da14d6e0b090.png)

#### Contributing README

![screencapture-github-Shopify-themekit-blob-60be6ebb4a8b0633f66091bf0720a8659cfd30ca-CONTRIBUTING-md-2021-01-23-14_38_16](https://user-images.githubusercontent.com/65373971/105616034-bb0c1a00-5d88-11eb-899d-a17c7c95db04.png)

#### Releasing README

![screencapture-github-Shopify-themekit-blob-93d8d6f0698fd1dbd8f262592ad3a99c203d5681-RELEASING-md-2021-02-02-14_14_58](https://user-images.githubusercontent.com/65373971/106670067-6a9f7400-6561-11eb-9f02-72675fad1273.png)

andyw8 commented on RELEASING.md:
```diff
@@ -12,7 +12,7 @@
   - Include links to the Pull Requests that introduced these changes
-- Update the [documentation website](https://shopify.github.io/themekit/)
+- Update the [documentation website](https://shopify.dev/tools/theme-kit)
   - run `make serve_docs`
```
As we're moving the docs to shopify.dev then `make serve_docs` would no longer apply (and we should probably remove this script from the repo).

andyw8 replied:
Also, we should think about the workflow for how doc changes are deployed alongside with new releases. Perhaps the first step should be to open the docs PR on shopify-dev, and get it approved. Then once the new Theme Kit version is released, the docs PR can be merged and deployed right away.

tanema reviewed (approved):
This looks good with a few comments. If this is full switch over then we will be able to delete the docs directory however that has some caveats.
- We need to update the docs every time there is a release. Should there be a downloads-only page that we still have on the github pages to make this easier? We can make it minimal and have links all directing back to the shopify.dev docs 
- There are static files like the python install script in the docs directory so we will need to find somewhere else for that to go.
- The changes to the command help output will require a release for it to update so we will need to coordinate this rollout

andyw8 commented:
> We need to update the docs every time there is a release. 

Is this only due to the Manual Installation links? I wonder if we could use the GitHub releases page for that instead

mcvinci commented:
Thanks for the feedback, @andyw8 and @tanema -- really appreciate you both taking a look at this work!

A few updates since you last saw this PR: 😄 

- I’ve removed instances of `make docs_serve` in this [commit](https://github.com/Shopify/themekit/pull/887/commits/bfb9eafc5223b7d7cf48dd4ed6e65c28a2d263ae). 
- I've moved the python script to `cmd/scripts` and deleted the entire `/docs` folder now. If I've accidentally wiped out something important, just let me know, and I shall restore it once more! If there's a better place for that python script, I'm happy to move to a more appropriate location, too.
- I've updated the "Releasing Theme Kit" README (`RELEASING.md`) to account for workflow changes, especially in relation to updating the docs and download links for Theme Kit. Open to your thoughts / suggestions! 
- I think @gfscott is aiming to have the Theme Kit docs updates live on February 22nd. Should we aim for a rollout on or near that date so that the command help output can be updated?

mcvinci commented:
@tanema Just wanted to check in on the timing of merging this PR, considering the changes to the command help output will require a release. Should we merge this PR this week?

tanema commented on README.md:
```diff
@@ -12,22 +12,83 @@
+- [Windows](https://shopify.dev/tools/theme-kit/getting-started#windows)
 
-Please see the [contributing guidlines](https://github.com/Shopify/themekit/blob/master/CONTRIBUTING.md)
+### Install Theme Kit manually
```
Should we make sure they only do this if they know what they are doing? Otherwise we should be redirecting them to the shopify.dev site for package manager installation. I am worried that this takes up too much space and developers will start to think that this is how you install themekit in general.

Lets take out the table with the binary downloads because I think we will move them to the github releases, and maybe move the manual installation instructions into a smaller md file that is off the beaten path.

Also can the AUR package not live with the rest of shopify.dev installation instructions?

mcvinci replied:
Thanks, @tanema! Great feedback. I've lifted the manual installation instructions out of the main README now, created a new `MANUAL_INSTALL.md` file to contain that content, removed the table with the binary downloads, and linked out to the GitHub releases page. I've also included a note to redirect them to the command line installation instructions on shopify.dev (to further encourage the preferred / recommended installation method).

In terms of the `AUR` package, I think we want to provide [the green path](https://github.com/Shopify/shopify-dev/pull/5070#discussion_r565603966) on shopify.dev. Any alternate methods, including manual installation steps, and accessing Theme Kit through community packages, can probably remain in the new `MANUAL_INSTALL.md` topic in the GitHub repo.

tanema commented on RELEASING.md:
```diff
@@ -1,30 +1,54 @@
+
+1. In GitHub, create a new release for the tag.
+2. Include a brief summary of all the changes pertaining to the release.
+3. Include links to the pull requests that introduced the changes.
```
Because of my previous comment, we should add here to upload the binaries, and put the md5s in the description.

tanema reviewed (commented):
Okay, I added some follow up comments about installation instructions. Also we should coodinate the release with the shopify.dev change because otherwise the links to `install.py` will be broken and there will be no docs for themekit until shopify.dev supplies them.

Also I think the install.py should live in a root scripts folder and not in cmd.

Sorry about the delay on my feedback

mcvinci commented:
Thanks for taking another look through, @tanema! I've moved `install.py` to a root `/scripts` folder and also addressed your additional feedback. 

If all looks good, I'm ready to merge this PR and the [shopify-dev PR](https://github.com/Shopify/shopify-dev/pull/5070) anytime. Let me know what timing works best to deploy the PRs.

tanema commented:
The best timing is at the same time as the shopify.dev changes. It is fine if the CLI lags because those who dont update will have that problem for a long time anyways

mcvinci commented:
@tanema Sounds good! I'll merge both PRs at the same time in a bit.

mcvinci commented:
/shipit
